### DISCUSSION TITLE
What's the future of `evaluate` in `forecasting.model_evaluation`?

### DISCUSSION THREAD

**Discussion by topher-lo** [alan-turing-institute/sktime#3452] [Q&A]
Hello sktime community! 👋 Got a question about "benchmarking" in sktime. It seems that there is a lot of work in building out the `benchmarking` subpkg in #2804 with kotsu. What is the future of the `evaluate` function in `forecasting/model_evaluation/_functions.py`? From first glance, it looks like the benchmarking package covers everything that `evaluate` does and more.

Seeking clarification as I have both joblib and "native" Dask code for parallelizing the `evaluate` function. Was wondering if it's worth contributing  or whether I should join the more recent discussion on "benchmarking"?

**Comment by fkiraly** [comment]
> Hello sktime community! 👋 Got a question about "benchmarking" in sktime. It seems that there is a lot of work in building out the `benchmarking` subpkg in #2804 with kotsu. What is the future of the `evaluate` function in `forecasting/model_evaluation/_functions.py`? From first glance, it looks like the benchmarking package covers everything that `evaluate` does and more.

That is an excellent question.

There are actually two important reverse dependencies:

* simple benchmarking workflow that a user might run to evaluate a forecaster or multiple, on one or multiple benchmark data sets
* the grid search estimators like `ForecastingGridSearchCV`

The first we will probably like to replace by sth `kotsu` based (once @DBCerigo gets back to it 😄 )

One could, in theory, also replace occurrence the second by `kotsu` based evaluation but that might tie the code to too many locations so is probably not good design-wise (cohesion, SOLID, and such). I.e., in the grid search, we still would like sth like a small and crisp `evaluate`.
@aiwalter, penny for your thoughts?

Based on that, I think joblib/dask versions of `evaluate` would be very useful, since they would give rise to joblib/dask parallelized grid search!
